nodejs/kubernetesx/kx.ts

```diff
@@ -278,16 +282,19 @@ export class Deployment extends k8s.apps.v1.Deployment {
                     }
                 });
                 return {
-                    ...args,
-                    ports: args.ports || ports,
+                    ...spec,
+                    ports: spec.ports || ports,
                     selector: this.spec.selector.matchLabels,
                     // TODO: probably need to unwrap args.type in case it's a computed value
-                    type: args && args.type as string,
+                    type: spec && spec.type as string,
                 };
             });
 
-        return new Service(this.name, {
-            metadata: { namespace: this.metadata.namespace },
+        return new Service(String(metadata?.name) ?? this.name, {
+            metadata: {
+                namespace: this.metadata.namespace,
+                name: metadata?.name ?? this.metadata.name,
```

Comment:
By setting the name here with a default, how will this impact existing users of the library?